### PR TITLE
JSUI-2721 Refactor DynamicFacets custom analytics data

### PR DESCRIPTION
**Pull request by ThibodeauJF**
Look at the AnalyticsActionListMeta.ts file to make sure everything is there.

https://coveord.atlassian.net/browse/JSUI-2721

Screen Shot 2019-11-20 at 4 23 36 PM

[Deploy](https://dashboard.heroku.com/pipelines/a3535101-5bbf-4a5b-a909-47fcf8c9f149)

### CODE REVIEW

File: src/ui/Analytics/AnalyticsActionListMeta.ts [hunk 1]
```diff
@@ -809,24 +685,69 @@ export var analyticsActionCauseList = {
     type: 'categoryFacet'
   },
   /**
-   * The custom event that gets logged when an end-user expands a dynamic facet to see additional values.
+   * Identifies the search event that gets logged when a dynamicFacet check box is selected and the query is updated.
```
Comment (ThibodeauJF) [inline src/ui/Analytics/AnalyticsActionListMeta.ts]: moved them up in the file so it's more readable

**Comment by ThibodeauJF** [comment]
@fbeaudoincoveo Sorry for the big review from you but I removed a lot of copy-pasted info from the AnalyticsActionListMeta.ts file

File: src/ui/Analytics/AnalyticsActionListMeta.ts [hunk 2]
```diff
@@ -468,11 +374,8 @@ export var analyticsActionCauseList = {
    * `actionCause`: `'documentTag'`
    * `actionType`: `'document'`
    *
-   * Logging an event with this actionType also adds the following key-value pairs in the custom data property of the Usage Analytics HTTP service request.
-   * `"facetId":`: <correspondingFacetId>
-   * `"facetValue":`: <correspondingFacetValue>
-   * `"facetTitle":`: <correspondingFacetTitle>
-   * `"facetField":`: <correspondingFacetField>
+   * The required properties of an [`IAnalyticsFacetMeta`](@link IAnalyticsFacetMeta) object are added as custom data
```
Comment (fbeaudoincoveo) [inline src/ui/Analytics/AnalyticsActionListMeta.ts]: Lovely :)

File: src/ui/Analytics/AnalyticsActionListMeta.ts (hunk 1 above)
Comment (fbeaudoincoveo) [inline src/ui/Analytics/AnalyticsActionListMeta.ts]: when a dynamicFacet -> when a DynamicFacet

Comment (fbeaudoincoveo) [reply src/ui/Analytics/AnalyticsActionListMeta.ts]: Same comment for other similar occurrences

File: src/ui/Analytics/AnalyticsActionListMeta.ts [hunk 3]
```diff
@@ -809,24 +685,69 @@ export var analyticsActionCauseList = {
...
+   *
+   * The required and optional properties of an [`IAnalyticsFacetMeta`](@link IAnalyticsFacetMeta)
+   * object are added as custom data when logging a usage analytics event matching this `actionCause`/`actionType`.
+   */
+  dynamicFacetDeselect: <IAnalyticsActionCause>{
+    name: 'facetDeselect',
+    type: 'dynamicFacet'
+  },
+  /**
+   * Identifies the search event that gets logged when the **Clear** button of the DynamicFacet is selected.
```
Comment (fbeaudoincoveo) [inline src/ui/Analytics/AnalyticsActionListMeta.ts]: Suggestion:

selected -> clicked

Comment (fbeaudoincoveo) [reply src/ui/Analytics/AnalyticsActionListMeta.ts]: Same comment for other similar occurrences

**Pull request merged**